### ISSUE TITLE
[CI] `linux://doc/source/ray-air/examples:upload_to_wandb` is failing/flaky on master.

### ISSUE THREAD

**Issue by cadedaniel** [ray-project/ray#29720]
- 182744bbd151c166b8028355eae12a5da63fb3cc FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/736#018413b7-de85-432b-ae0d-0e74af605fa6)
- 37de814a8598e0ea3dea23d5ae0caf9df54fa0e6 FAILED [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/734#0184130c-88fc-477d-a499-4a506a1341ce)
- 875193ca8d4720a7215aad2a275851f5157674c2 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/733#018412c7-9ce7-42fb-aaa1-d47f01a4c751)
- 824c79011e32fa4a0e9c409ef55a62a05fb70da8 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/732#018412aa-e3c8-4960-9275-fe968befbf85)
- 161567dbd93f94bef941c17779d3880f310a0373 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/730#01841286-1a93-4247-b0f3-7fb4a6dcafcf)
- e5e3f30dd8dcfe868313e5c3ff5793b21837b6ef FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/729#018411e9-43c4-40d5-b606-84fcfed215c3)
- 6e1ac50716663f5d35bdb5f2bf9b1e88e5b7a3f5 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/728#018411d1-0acb-43cd-89e4-e62fb217df7a)
- ee4b23a91008f6784643ba3b11b7a1b0e13c434b FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/727#018411c8-5b35-4586-b87c-873bc24eb0d6)
- 8b4283fb3da524c63a41f34a5221e762a3dc1587 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/726#01841131-5d2f-456a-87d9-f7a52fdab9db)
- a4d5494d46baf89df50b6cc1914b0f479eb4283f FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/724#018410c5-2a2b-416c-b33e-f96574dc545a)
- e464bf07af9f6513bf71156d1226885dde7a8f46 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/718#01841048-5365-4e55-bd18-b6fc95df4a54)
- 8e3e741d9b94a2694e789d4bcdff16e4c27a501c FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/708#01840f3b-1eb0-4302-9623-e821c4a0bc84)
- aa9a53a3e5cf8a337871920daf9bc65c090d9c0d FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/703#01840d0b-9f01-4a47-8c13-26dddce15980)
- d4ea9fbf0a84f668a2f3935f9efc3386dd08691d FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/690#01840c1c-b5d7-4e1c-bf33-7e07c1382986)
- 160c3cfa8959d133088bc56ecf9443351c17a558 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/689#01840be2-2369-4653-8b63-82dccc39a681)
- f021621439f1e805131a63577f6de4d1b675fabf FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/680#01840ae0-1e6a-4981-857c-d3c4ff141eba)
- 7243d56185d4307f788b5cae1a259907492c7bae FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/679#01840aa6-843e-4763-aadb-55702c202c71)
- 1c82427c2e271918f0963dcf3b02069923ec571a FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/668#0183fcd2-d673-414a-8f07-1f7dfaafc676)
- 48805211c6f560281f4ab00fbb7a2633b5001e2b FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/665#0183fc68-5f39-4ea5-8265-8173536fd7fc)
- 2c032ebe0ce8e625796491030a759e5e01edbef9 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/636#0183f6c8-0700-492d-aaec-2f2fde3b2515)
- fcc0086154154847af9a26b8e09b60d678274ff6 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/627#0183f65c-0d10-49a7-aad3-9b8326f54c72)
- 25390ff98dd734616e619fd378224eca861e0d00 FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/625#0183f5bc-7b08-4f61-81d2-5ed6af84208a)
- 69785334c45df0cddd9983e6aec552895f648d8a FLAKY [Buildkite :book: Doc examples with authentication ](https://buildkite.com/ray-project/oss-ci-build-branch/builds/608#0183f26c-e9b3-476f-8d00-bf515218ef2e)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://doc/source/ray-air/examples:upload_to_wandb-END
....

**Comment by bveeramani** [comment]
Looks like this test has only failed once in the past 100 commits, so I'm going to deprioritize this issue. I'll try to get back to it as soon as I can.

![image](https://user-images.githubusercontent.com/26107013/198246978-8f4225ff-2188-41a0-a7e5-a3261fbad433.png)

**Comment by cadedaniel** [comment]
I'm not sure what priority we should assign, but I created this issue because it is flaky (gives bad signal). Do we have a good prioritization guideline for fixing flaky tests? cc @jjyao

**Comment by jjyao** [comment]
For new flaky tests, they should be P0. For backlog flaky tests, we should try to fix them over the time. Our end goal is no flaky tests.

**Comment by bveeramani** [comment]
> For new flaky tests, they should be P0. For backlog flaky tests, we should try to fix them over the time. Our end goal is no flaky tests.

Looks like this has been flaky for some time. That said, I'll try to fix it this week.

I did some preliminary research -- looks like we're sending invalid requests to W&B. Not sure why that would cause a timeout, though.

```
2022-10-26 01:13:44,401	WARNING services.py:1748 -- WARNING: The object store is using /tmp instead of /dev/shm because /dev/shm has only 2684354560 bytes available. This will harm performance! You may be able to free up space by deleting files in /dev/shm. If you are inside a Docker container, you can increase /dev/shm size by passing '--shm-size=4.56gb' to 'docker run' (or add it to the run_options list in a Ray cluster config). Make sure to set this to more than 30% of available RAM.
--
... # Unrelated init code

  | 2022-10-26 01:13:47,929	INFO wandb.py:146 -- Already logged into W&B.
  | wandb: Currently logged in as: kaifricke (use `wandb login --relogin` to force relogin)
  | wandb: wandb version 0.13.4 is available!  To upgrade, please run:
  | wandb:  $ pip install wandb --upgrade
  | wandb: Tracking run with wandb version 0.12.15
  | wandb: Run data is saved locally in /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/_tmp/311fa3355db541337f9787b11b255fa0/XGBoostTrainer_2022-10-26_01-13-47/XGBoostTrainer_7115b_00000_0_2022-10-26_01-13-48/wandb/run-20221026_011348-7115b_00000
  | wandb: Run `wandb offline` to turn off syncing.
  | wandb: Syncing run XGBoostTrainer_7115b_00000
  | wandb: ⭐️ View project at https://wandb.ai/kaifricke/ray_air_example
  | wandb: 🚀 View run at https://wandb.ai/kaifricke/ray_air_example/runs/7115b_00000
  | wandb: Adding directory to artifact (/root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/_tmp/311fa3355db541337f9787b11b255fa0/XGBoostTrainer_2022-10-26_01-13-47/XGBoostTrainer_7115b_00000_0_2022-10-26_01-13-48/checkpoint_000010)... Done. 0.1s
  | wandb: Adding directory to artifact (/root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/_tmp/311fa3355db541337f9787b11b255fa0/XGBoostTrainer_2022-10-26_01-13-47/XGBoostTrainer_7115b_00000_0_2022-10-26_01-13-48/checkpoint_000010)... Done. 0.1s
  | wandb: Waiting for W&B process to finish... (success).
  | wandb: ERROR Error uploading "/root/.cache/wandb/artifacts/obj/md5/95/01c2e9ef7e26ae275103af2b72e722": CommError, <Response [400]>
  | wandb: ERROR Uploading artifact file failed. Artifact won't be committed.
  | (_RemoteRayXGBoostActor pid=2116) [01:13:59] task [xgboost.ray]:140093507919248 got new rank 1
  | (_RemoteRayXGBoostActor pid=2117) [01:13:59] task [xgboost.ray]:139655575907792 got new rank 0
  | 2022-10-26 01:14:11,306	INFO tune.py:788 -- Total run time: 24.26 seconds (23.27 seconds for the tuning loop)

== Status ==
--
  | Current time: 2022-10-26 01:13:51 (running for 00:00:03.44)
  | Memory usage on this node: 2.9/15.3 GiB
  | Using FIFO scheduling algorithm.
  | Resources requested: 3.0/4 CPUs, 0/0 GPUs, 0.0/8.3 GiB heap, 0.0/4.15 GiB objects
  | Result logdir: /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/_tmp/311fa3355db541337f9787b11b255fa0/XGBoostTrainer_2022-10-26_01-13-47
  | Number of trials: 1/1 (1 RUNNING)
  | +----------------------------+----------+------------------+
  | \| Trial name                 \| status   \| loc              \|
  | \|----------------------------+----------+------------------\|
  | \| XGBoostTrainer_7115b_00000 \| RUNNING  \| 172.16.16.3:1988 \|
  | +----------------------------+----------+------------------+

... # Tune status updates continue until timeout

-- Test timed out at 2022-10-26 01:28:39 UTC --
```

**Comment by krfricke** [comment]
Should be solved by #30165
